state/indexer/block/kv/kv.go

```diff
@@ -318,6 +347,33 @@ LOOP:
 	return filteredHeights, nil
 }
 
+func (idx *BlockerIndexer) setTmpHeights(tmpHeights map[string][]byte, it dbm.Iterator, matchEvents bool) {
+	// If we return attributes that occur within the same events, then store the event sequence in the
+	// result map as well
+	if matchEvents {
+		eventSeq, _ := parseEventSeqFromEventKey(it.Key())
+		retVal := it.Value()
+		tmpHeights[string(append(retVal, byte(eventSeq)))] = it.Value()
```

Comment:
so what happens if there are more than 256 events in a height?

Reply:
I completely overlooked that in Go `string()` will parse this to the ASCII value of  the int and it will wrap around if we get values bigger than 256:/ I will figure out how to change this. Nice catch, and many thanks!

Reply:
Should be fixed now.